drift_dev/lib/src/analysis/resolver/file_analysis.dart

```diff
@@ -73,6 +73,14 @@ class FileAnalyzer {
               .whereType<DriftElement>()
               .followedBy(availableByDefault)
               .transitiveClosureUnderReferences()
+              .where((e){
```

Comment:
This where should be in the one from line 70, the private elements shouldn't even reach `transitiveClosureUnderReferences`.

Reply:
What I found, using the example case in this comment (and the project file in the issue report), is that the private classes are sneaking in via '.followedBy(availableByDefault)' - It can be moved above `transitiveClosureUnderReferences ` move it up one, but any further and it wont work anymore.

Reply:
> followedBy(availableByDefault)

Wait really? Do you have an example for that? `availableByDefault` only includes tables and views explicitly added to the `@DriftDatabase` annotation, it should be impossible to reference private tables with that right?

Reply:
I've added a new comment further down which addresses everything, including this (my mistake)